* Usage instructions
Firstly, install required libraries - for the python libs that I've used in my analysis you can just run this, however it installs more than what is needed:

#+BEGIN_SRC bash
pip install ./requirements.txt
#+END_SRC

** Remember to install Trimmomatic
E.g.
#+BEGIN_SRC bash
wget http://www.usadellab.org/cms/uploads/supplementary/Trimmomatic/Trimmomatic-0.39.zip
unzip Trimmomatic*
#+END_SRC


Obviously the variables being used for locations and such will need changed

** If you have access to an HPC running slurm

#+BEGIN_SRC bash
chmod +x ./run_HPC_analysis.sh
./run_HPC_analysis.sh
#+END_SRC

** Else

#+BEGIN_SRC bash

FASTQ="../arabidopsis_thaliana/seedling_data/raw/raw_data/"
FASTQ_R="../arabidopsis_thaliana/seedling_data/raw/raw_data/"
FASTQ_O="../arabidopsis_thaliana/seedling_data/fastq/"
FASTQ_O_R="../arabidopsis_thaliana/seedling_data/fastq/"
FORWARD_I="_1"
REVERSE_I="_2"
REPORTS="../arabidopsis_thaliana/seedling_data/reports"
TRIMMED_REPORTS="../arabidopsis_thaliana/seedling_data/trimmed_reports"
INDEX="../arabidopsis_thaliana/Reference/hisat2_index/arabidopsis.hisat2"
BAM="../arabidopsis_thaliana/seedling_data/bam"
GTF="../arabidopsis_thaliana/Reference/Arabidopsis_thaliana.TAIR10.43.gtf"
HTSEQ_COUNT="../arabidopsis_thaliana/seedling_data/htseq-count"
ADAPTORS="novogene.fa"


echo "Running pre qc"
python3 ./qc.py -f $FASTQ -o $REPORTS

#Perform adaptor removal
echo "Adaptor trimming"
python3 ./remove_adaptors.py -f ${FASTQ} -r ${FASTQ_R} -F ${FASTQ_O} -R ${FASTQ_O_R} -a ${ADAPTORS} -i ${FORWARD_I} -I ${REVERSE_I}

# Perform fastqc analysis
# -> reports and multiqc report post trimming
echo "Running post qc"
python3 ./qc.py -f $FASTQ -o $TRIMMED_REPORTS

# align the fastq to annotation
# fastq -> bam
echo "Performing align"
python3 ./align.py -f $FASTQ -r $INDEX -o $BAM -F $FORWARD_I -R $REVERSE_I

# bam -> counts
python3 ./count_genes.py -b $BAM -g $GTF -o $HTSEQ_COUNT


#+END_SRC
